authzed/api/v1/experimental_service.proto

```diff
@@ -136,3 +184,174 @@ message BulkExportRelationshipsResponse {
   Cursor after_result_cursor  = 1;
   repeated Relationship relationships = 2;
 }
+
+// Reflection types ////////////////////////////////////////////
+
+message ExperimentalReflectSchemaRequest {
+  Consistency consistency = 1;
+ 
+  // optional_filters defines optional filters that are applied in
+  // an OR fashion to the schema, before being returned
+  repeated ExpSchemaFilter optional_filters = 2;
+}
+
+message ExperimentalReflectSchemaResponse {
+  // definitions are the definitions defined in the schema.
+  repeated ExpDefinition definitions = 1;
+
+  // caveats are the caveats defined in the schema.
+  repeated ExpCaveat caveats = 2;
+
+  // read_at is the ZedToken at which the schema was read.
+  ZedToken read_at = 3;
+}
+
+message ExpSchemaFilter {
+  enum KindFilter {
+    KIND_FILTER_UNSPECIFIED = 0;
+    KIND_FILTER_DEFINITION = 1;
+    KIND_FILTER_CAVEAT = 2;
+    KIND_FILTER_RELATION = 3;
+    KIND_FILTER_PERMISSION = 4;
+  }
+
+  // optional_definition_name_match is a regex that is matched against the definition or caveat name.
+  // If not specified, will be ignored.
+  string optional_definition_name_match = 1;
+
+  // optional_relation_or_permission_name_match is a regex that is matched against the relation or permission name.
+  // If not specified, will be ignored.
+  string optional_relation_or_permission_name_match = 2;
+
+  // kind_filters is a list of kinds to filter on. If not specified, will be ignored. If multiple are specified,
+  // the filter will be applied in an OR fashion.
+  repeated KindFilter kind_filters = 3;
+}
+
+message ExpDefinition {
+  string name = 1;
+  string comment = 2;
```

Comment:
Food for thought: something that would be awesome would be to provide a bit more structure to comments than just a plain string. The use-case would be giving users the ability to add metadata to definitions so they can create a correlation between their application business domain and the schema.

```go
// resource represents the basic object of the business domain
// @xref: IResource
// @codeowner: joey
// @version: 1.1
definition resource { }
```

Then the API response could have a type which looks like:

```protobuf
message ExpDefinition {
  string name = 1;
  string comment = 2;
  map<string, string> annotations = 3;
}
```

Reply:
Its planned for the new schema support but not something I want to do at this time